astropy/coordinates/baseframe.py

```diff
@@ -1791,27 +1817,34 @@ def position_angle(self, other: BaseCoordinateFrame | SkyCoord) -> Angle:
         >>> c1.position_angle(c3).degree  # doctest: +FLOAT_CMP
         44.995636455344844
         """
-        return position_angle(*self._prepare_unit_sphere_coords(other))
+        return position_angle(*self._prepare_unit_sphere_coords(other, "ignore"))
 
-    def separation(self, other):
+    def separation(
+        self,
+        other: BaseCoordinateFrame | SkyCoord,
+        frame_mismatch: StrictnessControl = "warn",
```

Comment:
I feel like since Python 2 was dropped there hasn't been a compelling reason to *not* make every new arguments to public functions keyword-only (which matches how you're illustrating its use in docs anyway)
```suggestion
        *,
        frame_mismatch: StrictnessControl = "warn",
```

Reply:
Making positional-or-keyword arguments keyword-only is a backwards incompatible change, but making keyword-only arguments positional-or-keyword is backwards compatible. So I agree that it is better to err on the side of caution and make this keyword-only in the public API.